ereadingtool/ereadingtool
# Text rating

PR brings in rating feature that takes a vote cast by a student. It checks on the backend to confirm they're allowed to vote then returns a small amount of data to the frontend. The front will take this data, `textId, vote, rating` and update the corresponding text with the new rating.

**How to test**
After checking out the branch, navigate to the text search page and attempt to rate a text that has yet to be read. It should not be possible. Click the text and read it to completion or partially, either works. Afterwards, navigate back to that card on the text search page. You should now be able to rate it. 

Giving it a plus vote should add one to the rating, giving it a minus should subtract one. You can also undo your votes.

**Most affected files**
`text/views/api/text.py`
`web/src/Pages/Text/Search.elm`
